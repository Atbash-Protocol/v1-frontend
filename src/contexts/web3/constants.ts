export const USER_REJECTED = "User Rejected";
