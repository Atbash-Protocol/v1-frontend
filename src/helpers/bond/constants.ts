import { Networks } from '../../constants/blockchain';

export enum BondType {
    STABLE_ASSET,
    LP,
}

export interface BondAddresses {
    reserveAddress: string;
    bondAddress: string;
}

export interface NetworkAddresses {
    [Networks.MAINNET]: BondAddresses;
    [Networks.RINKEBY]: BondAddresses;
    [Networks.LOCAL]: BondAddresses;
}
