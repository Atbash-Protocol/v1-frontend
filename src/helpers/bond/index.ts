import { Networks } from '../../constants/blockchain';
import { LPBond, CustomLPBond } from './lp-bond';
import { StableBond, CustomBond } from './stable-bond';

import ETHIcon from '../../assets/tokens/ETH.svg';
import AvaxIcon from '../../assets/tokens/floof.png';
import BASHUSDCison from '../../assets/tokens/BASH-USDC.png';
import bashUSDTIcon from '../../assets/tokens/BASH-USDT.png';
import BashIcon from '../../assets/tokens/bash.png';
import DAIIcon from '../../assets/tokens/DAI.e.png';
import MimSdogIcon from '../../assets/tokens/SDOG-MIM.svg';
import avaxUsdceIcon from '../../assets/tokens/AVAX-USDCe.png';
import bashDAIIcon from '../../assets/tokens/BASH-DAI.png';
import BASHFLOOFIcon from '../../assets/tokens/BASH-FLOOF.png';

import { StableBondContract, LpBondContract, WavaxBondContract, StableReserveContract, LpReserveContract } from '../../abi';

import { BashDaiLpAddress, BashDaiBondAddress, DaiAddress, DaiBondAddress, BashAddress } from 'constants/addresses';

export const dai = new StableBond({
    name: 'DAI',
    displayName: 'BASH',
    bondToken: 'DAI',
    bondIconSvg: BashIcon,
    bondContractABI: StableBondContract,
    reserveContractAbi: StableReserveContract,
    // network addresses will come from imported deployments
    networkAddrs: {
        [Networks.RINKEBY]: {
            bondAddress: '',
            reserveAddress: '',
        },
        [Networks.LOCAL]: {
            bondAddress: '',
            reserveAddress: '',
        },
        [Networks.MAINNET]: {
            bondAddress: '',
            reserveAddress: '',
        },
    },
    isActive: true,
});

export const wavax = new CustomBond({
    name: 'FLOOF',
    displayName: 'FLOOF',
    bondToken: 'ETH',
    bondIconSvg: AvaxIcon,
    bondContractABI: WavaxBondContract,
    reserveContractAbi: StableReserveContract,
    networkAddrs: {
        [Networks.MAINNET]: {
            bondAddress: '0x472c18c4079eCb68629F4FbA1141172404BFEE9C',
            reserveAddress: '0xb31f66aa3c1e785363f0875a1b74e27b85fd66c7',
        },
        [Networks.LOCAL]: {
            bondAddress: '',
            reserveAddress: '',
        },
        [Networks.RINKEBY]: { bondAddress: '', reserveAddress: '' },
    },
    isActive: false,
});

export const BASHUSDC = new LPBond({
    name: 'bash_usdc_lp',
    displayName: 'BASH-USDC LP',
    bondToken: 'USDC',
    bondIconSvg: BASHUSDCison,
    bondContractABI: LpBondContract,
    reserveContractAbi: LpReserveContract,
    // network addresses will come from imported deployments
    networkAddrs: {
        [Networks.MAINNET]: {
            bondAddress: '0x90A08fdF9f433954930f19E97FE9A1B0bDBf5C5f',
            reserveAddress: '0x425c45adfb53861e5db8f17d9b072ab60d4404d8',
        },
        [Networks.LOCAL]: {
            bondAddress: '',
            reserveAddress: '',
        },
        [Networks.RINKEBY]: { bondAddress: '', reserveAddress: '' },
    },
    lpUrl: 'https://www.traderjoexyz.com/#/pool/0x130966628846BFd36ff31a822705796e8cb8C18D/0x7d1232b90d3f809a54eeaeebc639c62df8a8942f',
    isActive: false,
});

export const BASHDAI = new LPBond({
    name: 'bash_dai_lp',
    displayName: 'BASH-DAI LP',
    bondToken: 'DAI',
    bondIconSvg: bashDAIIcon,
    bondContractABI: LpBondContract,
    reserveContractAbi: LpReserveContract,
    // network addresses will come from imported deployments
    networkAddrs: {
        [Networks.MAINNET]: {
            bondAddress: '',
            reserveAddress: '',
        },
        [Networks.LOCAL]: {
            bondAddress: '',
            reserveAddress: '',
        },
        [Networks.RINKEBY]: {
            bondAddress: '',
            reserveAddress: '',
        },
    },
    lpUrl: `https://app.uniswap.org/#/swap?inputCurrency=0x6b175474e89094c44da98b954eedeac495271d0f&outputCurrency=0x5646077e2779a4e8607a99257bd76f627a262d86&chain=mainnet`, // sushi rinkeby: "https://app.sushi.com/add/0x6C538aDf35d1927497090e6971Fc46D8ed813dF6/0xdc7B08BB2AbcE1BA5b82509115F3fb7358E412aB",
    isActive: true,
});

// export const avaxUsdce = new CustomLPBond({
//     name: "usdce_avax_lp",
//     displayName: "USDC.e-AVAX LP",
//     bondToken: "AVAX",
//     bondIconSvg: avaxUsdceIcon,
//     bondContractABI: LpBondContract,
//     reserveContractAbi: LpReserveContract,
//     networkAddrs: {
//         [Networks.AVAX]: {
//             bondAddress: "0x5E831EF7B5B2824B6F327055d0193Ecf8dCAA480",
//             reserveAddress: "0xa389f9430876455c36478deea9769b7ca4e3ddb1",
//         },
//     },
//     lpUrl: "https://www.traderjoexyz.com/#/pool/0xa7d7079b0fead91f3e65f86e8915cb59c1a4c664",
//     isActive: true,
// });

// duplicate?
// export const bashDai = new StableBond({
//     name: "bash_dai_minting",
//     displayName: "BASH-DAI LP",
//     bondToken: "DAI",
//     bondIconSvg: bashDAIIcon,
//     bondContractABI: StableBondContract,
//     reserveContractAbi: StableReserveContract,
//     // network addresses will come from imported deployments
//     networkAddrs: {
//         [Networks.RINKEBY]: {
//             bondAddress: "",
//             reserveAddress: "",
//         },
//         [Networks.LOCAL]: {
//             bondAddress: "",
//             reserveAddress: "",
//         },
//         [Networks.MAINNET]: { bondAddress: "", reserveAddress: "" },
//     },
//     isActive: true,
// });

export const bashFloof = new LPBond({
    name: 'bash_floof_minting',
    displayName: 'BASH-FLOOF LP',
    bondToken: 'FLOOF',
    bondIconSvg: BASHFLOOFIcon,
    bondContractABI: LpBondContract,
    reserveContractAbi: LpReserveContract,
    networkAddrs: {
        [Networks.MAINNET]: {
            bondAddress: '0x90A08fdF9f433954930f19E97FE9A1B0bDBf5C5f',
            reserveAddress: '0x781655d802670bba3c89aebaaea59d3182fd755d',
        },
        [Networks.LOCAL]: {
            bondAddress: '',
            reserveAddress: '',
        },
        [Networks.RINKEBY]: { bondAddress: '', reserveAddress: '' },
    },
    lpUrl: 'https://traderjoexyz.com/#/pool/AVAX/0x130966628846bfd36ff31a822705796e8cb8c18d',
    isActive: false,
});

// export const mimSdog = new LPBond({
//     name: "mim_sdog_lp",
//     displayName: "SDOG-MIM LP",
//     bondToken: "MIM",
//     bondIconSvg: MimSdogIcon,
//     bondContractABI: LpBondContract,
//     reserveContractAbi: LpReserveContract,
//     networkAddrs: {
//         [Networks.AVAX]: {
//             bondAddress: "0x0d71d9616f6844f12f1ce1516aa67817754517ff",
//             reserveAddress: "0xa3f1f5076499ec37d5bb095551f85ab5a344bb58",
//         },
//     },
//     lpUrl: "https://traderjoexyz.com/#/pool/0x130966628846BFd36ff31a822705796e8cb8C18D/0xdE9E52F1838951e4d2bb6C59723B003c353979b6",
// });

export const bashUSDT = new CustomLPBond({
    name: 'bash_USDT_lp',
    displayName: 'BASH-USDT LP',
    bondToken: 'USDT',
    bondIconSvg: bashUSDTIcon,
    bondContractABI: LpBondContract,
    reserveContractAbi: LpReserveContract,
    networkAddrs: {
        [Networks.MAINNET]: {
            bondAddress: '0x90A08fdF9f433954930f19E97FE9A1B0bDBf5C5f',
            reserveAddress: '0xa3d2cfe49df9d1ea0dc589b69252e1eddc417d6d',
        },
        [Networks.LOCAL]: {
            bondAddress: '',
            reserveAddress: '',
        },
        [Networks.RINKEBY]: { bondAddress: '', reserveAddress: '' },
    },
    lpUrl: 'https://traderjoexyz.com/#/pool/AVAX/0x7d1232b90d3f809a54eeaeebc639c62df8a8942f',
    isActive: false,
});

// export default [dai, wavax, BASHUSDC, bashUSDT, bashDai];
export default [dai, BASHDAI];
