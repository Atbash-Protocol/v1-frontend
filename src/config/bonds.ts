import { Networks } from "constants/blockchain";
import { BondType } from "helpers/bond/constants";
import { BondConfig, BondProviderEnum } from "lib/bonds/bonds.types";
import { ActiveTokensEnum } from "./tokens";

import MimIcon from "assets/tokens/BASH-DAI.png";
import DAIIcon from "assets/tokens/DAI.e.png";

const BASH_DAI: BondConfig = {
    name: "bash_dai_lp",
    displayName: "BASH-DAI LP",
    token: ActiveTokensEnum.DAI,
    iconPath: MimIcon,
    lpProvider: BondProviderEnum.UNISWAP_V2, // sushi rinkeby: "https://app.sushi.com/add/0x6C538aDf35d1927497090e6971Fc46D8ed813dF6/0xdc7B08BB2AbcE1BA5b82509115F3fb7358E412aB",
    type: BondType.LP,
    addresses: {
        [Networks.MAINNET]: {
            bondAddress: "",
            reserveAddress: "",
        },
        [Networks.LOCAL]: {
            bondAddress: "0x7bc06c482DEAd17c0e297aFbC32f6e63d3846650",
            reserveAddress: "0x908B40ED87FCA620B101B3Cba2B4C640D11eF016",
        },
        [Networks.RINKEBY]: {
            bondAddress: "0xcE24D6A45D5c59D31D05c8C278cA3455dD6a43DA",
            reserveAddress: "0x26DF06b47412dA76061ddA1fD9fe688A497FB88b", // uniswapv2-"0xC35F84DBd48fcB0467ac3Ee2C4e37D848B8d3173",
        },
    },
    isActive: false,
};

const StableDAI: BondConfig = {
    name: "DAI",
    displayName: "DAI",
    token: ActiveTokensEnum.DAI,
    iconPath: DAIIcon,
    lpProvider: BondProviderEnum.UNISWAP_V2, // sushi rinkeby: "https://app.sushi.com/add/0x6C538aDf35d1927497090e6971Fc46D8ed813dF6/0xdc7B08BB2AbcE1BA5b82509115F3fb7358E412aB",
    type: BondType.StableAsset,
    addresses: {
        [Networks.RINKEBY]: {
            bondAddress: "0xd0D5024c723c4F047C7626FBA18E7797A4E50a12",
            reserveAddress: "0xdc7B08BB2AbcE1BA5b82509115F3fb7358E412aB", // DAI
        },
        [Networks.LOCAL]: {
            bondAddress: "0x4ed7c70F96B99c776995fB64377f0d4aB3B0e1C1",
            reserveAddress: "0xDc64a140Aa3E981100a9becA4E685f962f0cF6C9", // DAI
        },
        [Networks.MAINNET]: {
            bondAddress: "",
            reserveAddress: "",
        },
    },
    isActive: true,
};

export const BONDS = [BASH_DAI, StableDAI];
