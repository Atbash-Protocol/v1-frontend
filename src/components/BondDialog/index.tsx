import { useEffect } from 'react';

import { Box, Dialog, DialogContent, DialogTitle, Divider, Grid, Typography } from '@mui/material';
import { t } from 'i18next';
import { useDispatch, useSelector } from 'react-redux';
import { useHistory } from 'react-router-dom';

import BondLogo from 'components/BondLogo';
import Loader from 'components/Loader';
import MenuMetric from 'components/Metrics/MenuMetric';
import { theme } from 'constants/theme';
import { useWeb3Context } from 'contexts/web3/web3.context';
import { formatUSD } from 'helpers/price-units';
import { LPBond } from 'lib/bonds/bond/lp-bond';
import { StableBond } from 'lib/bonds/bond/stable-bond';
import { selectFormattedReservePrice } from 'store/modules/app/app.selectors';
import { selectBondInstance, selectBondMetrics } from 'store/modules/bonds/bonds.selector';
import { calcBondDetails, getBondTerms, getTreasuryBalance, loadBondBalancesAndAllowances } from 'store/modules/bonds/bonds.thunks';
import { BondMetrics } from 'store/modules/bonds/bonds.types';
import { RootState } from 'store/store';
import BondPurchase from 'views/Bond/actions/BondPurchase';
import BondDetailsMetrics from 'views/Bond/BondMetrics';

const BondDetails = ({ open, bondID, bond, metrics }: { open: boolean; bondID: string; bond: LPBond | StableBond; metrics: BondMetrics }) => {
    const history = useHistory();
    const dispatch = useDispatch();

    const {
        state: { signer, signerAddress, networkID },
    } = useWeb3Context();

    const onBackdropClick = () => history.goBack();

    const bashPrice = useSelector(selectFormattedReservePrice);
    console.log('here', dispatch);

    useEffect(() => {
        if (!metrics?.terms) {
            console.log('here2');
            dispatch(getBondTerms(bondID));
        }
    }, [metrics?.terms]);

    console.log('bdetails', dispatch, calcBondDetails);

    useEffect(() => {
        if (bondID) {
            dispatch(calcBondDetails({ bondID, value: 0 }));

            if (signer && signerAddress) {
                dispatch(loadBondBalancesAndAllowances({ address: signerAddress || '', bondID }));
            }
        }
    }, [bondID, signer, signerAddress]);

    useEffect(() => {
        if (networkID) {
            dispatch(getTreasuryBalance({ networkID }));
        }
    }, [networkID]);

    //TODO: Add the custom settings : Slippage & Recipient address

    return (
        <Dialog
            {...{
                onBackdropClick,
                open,
                maxWidth: 'sm',
                fullWidth: true,
                PaperProps: { sx: { background: theme.palette.cardBackground.light, color: theme.palette.primary.dark } },
            }}
            sx={{
                p: 2,
                [theme.breakpoints.up('xs')]: {
                    p: 0,
                },
                backdropFilter: 'blur(10px)',
            }}
        >
            <DialogTitle sx={{ display: 'flex', alignItems: 'center', justifyContent: 'center', gap: theme.spacing(1) }}>
                <BondLogo bondLogoPath={bond.bondOptions.iconPath} isLP={bond.isLP()} />

                <Typography variant="body1">{bond.bondOptions.displayName}</Typography>
            </DialogTitle>
            <DialogContent>
                <Box>
                    <Grid container item xs={12} mb={4}>
                        <Grid item xs={12} sm={6}>
                            <MenuMetric key={'treasuryBalance'} metricKey={t('TreasuryBalance')} value={formatUSD(metrics.treasuryBalance || 0, 2)} />
                        </Grid>

                        <Grid item xs={12} sm={6}>
                            <MenuMetric key={'BashPrice'} metricKey={t('BASHPrice')} value={bashPrice} />
                        </Grid>
                    </Grid>
                </Box>

                {metrics.allowance !== null && (
                    <Box>
                        <BondPurchase bondID={bondID} />
                        <Divider variant="fullWidth" textAlign="center" sx={{ borderColor: theme.palette.primary.light, marginBottom: theme.spacing(2) }} />
                        <BondDetailsMetrics bondMetrics={metrics} />
                    </Box>
                )}
            </DialogContent>
        </Dialog>
    );
};

const BondDialogLoader = ({ open, bondID }: { open: boolean; bondID: string }) => {
    const metrics = useSelector((state: RootState) => selectBondMetrics(state, bondID));
    const bond = useSelector((state: RootState) => selectBondInstance(state, bondID));

    if (!metrics || !bond) return <Loader />;

    console.log('bond dialog', metrics, bond);

    return <BondDetails open={open} bondID={bondID} bond={bond} metrics={metrics} />;
};

export default BondDialogLoader;
