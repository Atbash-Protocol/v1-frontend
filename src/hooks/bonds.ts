import { useMemo } from 'react';

import { useSelector } from 'react-redux';

import { BondItem } from 'store/modules/bonds/bonds.types';
import { IReduxState } from 'store/slices/state.interface';

// Slaps together bond data within the account & bonding states
const useBonds = () => {
    const bonds = useSelector<IReduxState, BondItem[]>(state => Object.values(state.bonds.bonds));

    const mostProfitableBonds = useMemo(
        () =>
            bonds.sort((bond1, bond2): number => {
                if (bond1.metrics.bondDiscount === null || bond2.metrics.bondDiscount === null) return 0;

                return bond1.metrics.bondDiscount > bond2.metrics.bondDiscount ? 0 : 1;
            }),
        [bonds],
    );

    return {
        bonds,
        mostProfitableBonds,
    };
};

export default useBonds;

export const selectBondReady = (bond: BondItem) => {
    return Object.values(bond.metrics).some(metric => metric !== null);
};

export const useBondPurchaseReady = () => {
    const metrics = useSelector<IReduxState, boolean>(state => state.main.metrics.reserves !== null);

    const marketReady = useSelector<IReduxState, boolean>(state => state.markets.markets.dai !== null);

    return metrics && marketReady;
};
