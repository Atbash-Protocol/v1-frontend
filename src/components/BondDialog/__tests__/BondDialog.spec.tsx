import { configureStore } from '@reduxjs/toolkit';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import * as ReactReduxModule from 'react-redux';

import { Web3Context } from 'contexts/web3/web3.context';
import { BondType } from 'helpers/bond/constants';
import * as BondHookModule from 'hooks/bonds';
import { BondOptions } from 'lib/bonds/bond/bond';
import { LPBond } from 'lib/bonds/bond/lp-bond';
import { BondProviderEnum } from 'lib/bonds/bonds.types';
import mainReducer from 'store/modules/app/app.slice';
import BondsReducer from 'store/modules/bonds/bonds.slice';
import * as BondsReduxModule from 'store/modules/bonds/bonds.thunks';
import marketReducer from 'store/modules/markets/markets.slice';
import transactionReducer from 'store/modules/transactions/transactions.slice';

import BondDialog from '..';

function renderComponent(component: JSX.Element, contextState?: any) {
    return render(
        <Web3Context.Provider value={contextState}>
            <Provider
                store={configureStore({
                    reducer: {
                        main: mainReducer,
                        markets: marketReducer,
                        bonds: BondsReducer,
                        transactions: transactionReducer,
                    },
                    preloadedState: {
                        bonds: {
                            metrics: {},
                            bondQuote: {
                                interestDue: 123,
                            },
                        },
                    } as any,
                })}
            >
                {component}
            </Provider>
            ,
        </Web3Context.Provider>,
    );
}

describe('BondDialog', () => {
    const bondOptions: BondOptions = {
        name: 'bond',
        displayName: 'bondToDisplay',
        token: 'token',
        iconPath: 'path',
        lpProvider: BondProviderEnum.UNISWAP_V2,
        type: BondType.LP,
        networkID: 2,
        isActive: true,
    };

    const testBond = {
        bondInstance: new LPBond(bondOptions),
        metrics: {
            reserves: 0,
        },
        terms: {},
        bonds: {
            loading: false,
        },
    };

    it('renders', () => {
        jest.spyOn(BondHookModule, 'useBondPurchaseReady').mockReturnValue(true);
        const { container } = renderComponent(<BondDialog bond={testBond as any} open={true} />, { state: { signer: 'signer', signerAddress: 'signerAddress' } });
        expect(container).toMatchSnapshot();
    });

    it('dispatches the actions', () => {
        jest.spyOn(ReactReduxModule, 'useDispatch').mockReturnValue(jest.fn());
        jest.spyOn(BondHookModule, 'useBondPurchaseReady').mockReturnValue(true);
        jest.spyOn(BondHookModule, 'selectBondReady').mockReturnValue(false);
        const getTermsSpy = jest.spyOn(BondsReduxModule, 'getBondTerms').mockReturnThis();
        const calcBondDetailsSpy = jest.spyOn(BondsReduxModule, 'calcBondDetails').mockReturnThis();
        const loadBondBalancesAndAllowancesSpy = jest.spyOn(BondsReduxModule, 'loadBondBalancesAndAllowances').mockReturnThis();

        renderComponent(<BondDialog bond={testBond as any} open={true} />, { state: { signer: 'signer', signerAddress: 'signerAddress' } });

        expect(getTermsSpy).toHaveBeenCalledTimes(1);
        expect(getTermsSpy).toHaveBeenCalledWith(testBond);
        expect(calcBondDetailsSpy).toHaveBeenCalledTimes(1);
        expect(calcBondDetailsSpy).toHaveBeenCalledWith({ bond: testBond.bondInstance, value: 0 });
        expect(loadBondBalancesAndAllowancesSpy).toHaveBeenCalledTimes(1);
        expect(loadBondBalancesAndAllowancesSpy).toHaveBeenCalledWith({ address: 'signerAddress' });
    });
});
