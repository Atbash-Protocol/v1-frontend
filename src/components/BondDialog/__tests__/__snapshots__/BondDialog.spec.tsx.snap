// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BondDialog dispatches the actions 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style=""
  >
    <div
      id="WEB3_CONNECT_MODAL_ID"
    >
      <div
        class="sc-jSMfEi feQmWT web3modal-modal-lightbox"
        offset="0"
        opacity="0.4"
      >
        <div
          class="sc-gKXOVf gOSrSh web3modal-modal-container"
        >
          <div
            class="sc-iBkjds hbCoUo web3modal-modal-hitbox"
          />
          <div
            class="sc-ftvSup iabRxz web3modal-modal-card"
          >
            <div
              class="sc-eCYdqJ dFclSe web3modal-provider-wrapper"
            >
              <div
                class="sc-hKMtZM keqyKc web3modal-provider-container"
              >
                <div
                  class="sc-bczRLJ dtZuCG web3modal-provider-icon"
                >
                  <img
                    alt="WalletConnect"
                    src="data:image/svg+xml;base64,PHN2ZyBoZWlnaHQ9IjUxMiIgdmlld0JveD0iMCAwIDUxMiA1MTIiIHdpZHRoPSI1MTIiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgeG1sbnM6eGxpbms9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkveGxpbmsiPjxyYWRpYWxHcmFkaWVudCBpZD0iYSIgY3g9IjAlIiBjeT0iNTAlIiByPSIxMDAlIj48c3RvcCBvZmZzZXQ9IjAiIHN0b3AtY29sb3I9IiM1ZDlkZjYiLz48c3RvcCBvZmZzZXQ9IjEiIHN0b3AtY29sb3I9IiMwMDZmZmYiLz48L3JhZGlhbEdyYWRpZW50PjxnIGZpbGw9Im5vbmUiIGZpbGwtcnVsZT0iZXZlbm9kZCI+PHBhdGggZD0ibTI1NiAwYzE0MS4zODQ4OTYgMCAyNTYgMTE0LjYxNTEwNCAyNTYgMjU2cy0xMTQuNjE1MTA0IDI1Ni0yNTYgMjU2LTI1Ni0xMTQuNjE1MTA0LTI1Ni0yNTYgMTE0LjYxNTEwNC0yNTYgMjU2LTI1NnoiIGZpbGw9InVybCgjYSkiLz48cGF0aCBkPSJtNjQuNjkxNzU1OCAzNy43MDg4Mjk4YzUxLjUzMjgwNzItNTAuMjc4NDM5NyAxMzUuMDgzOTk0Mi01MC4yNzg0Mzk3IDE4Ni42MTY3OTkyIDBsNi4yMDIwNTcgNi4wNTEwOTA2YzIuNTc2NjQgMi41MTM5MjE4IDIuNTc2NjQgNi41ODk3OTQ4IDAgOS4xMDM3MTc3bC0yMS4yMTU5OTggMjAuNjk5NTc1OWMtMS4yODgzMjEgMS4yNTY5NjE5LTMuMzc3MSAxLjI1Njk2MTktNC42NjU0MjEgMGwtOC41MzQ3NjYtOC4zMjcwMjA1Yy0zNS45NTA1NzMtMzUuMDc1NDk2Mi05NC4yMzc5NjktMzUuMDc1NDk2Mi0xMzAuMTg4NTQ0IDBsLTkuMTQwMDI4MiA4LjkxNzU1MTljLTEuMjg4MzIxNyAxLjI1Njk2MDktMy4zNzcxMDE2IDEuMjU2OTYwOS00LjY2NTQyMDggMGwtMjEuMjE1OTk3My0yMC42OTk1NzU5Yy0yLjU3NjY0MDMtMi41MTM5MjI5LTIuNTc2NjQwMy02LjU4OTc5NTggMC05LjEwMzcxNzd6bTIzMC40OTM0ODUyIDQyLjgwODkxMTcgMTguODgyMjc5IDE4LjQyMjcyNjJjMi41NzY2MjcgMi41MTM5MTAzIDIuNTc2NjQyIDYuNTg5NzU5My4wMDAwMzIgOS4xMDM2ODYzbC04NS4xNDE0OTggODMuMDcwMzU4Yy0yLjU3NjYyMyAyLjUxMzk0MS02Ljc1NDE4MiAyLjUxMzk2OS05LjMzMDg0LjAwMDA2Ni0uMDAwMDEtLjAwMDAxLS4wMDAwMjMtLjAwMDAyMy0uMDAwMDMzLS4wMDAwMzRsLTYwLjQyODI1Ni01OC45NTc0NTFjLS42NDQxNi0uNjI4NDgxLTEuNjg4NTUtLjYyODQ4MS0yLjMzMjcxIDAtLjAwMDAwNC4wMDAwMDQtLjAwMDAwOC4wMDAwMDctLjAwMDAxMi4wMDAwMTFsLTYwLjQyNjk2ODMgNTguOTU3NDA4Yy0yLjU3NjYxNDEgMi41MTM5NDctNi43NTQxNzQ2IDIuNTEzOTktOS4zMzA4NDA4LjAwMDA5Mi0uMDAwMDE1MS0uMDAwMDE0LS4wMDAwMzA5LS4wMDAwMjktLjAwMDA0NjctLjAwMDA0NmwtODUuMTQzODY3NzQtODMuMDcxNDYzYy0yLjU3NjYzOTI4LTIuNTEzOTIxLTIuNTc2NjM5MjgtNi41ODk3OTUgMC05LjEwMzcxNjNsMTguODgyMzEyNjQtMTguNDIyNjk1NWMyLjU3NjYzOTMtMi41MTM5MjIyIDYuNzU0MTk5My0yLjUxMzkyMjIgOS4zMzA4Mzk3IDBsNjAuNDI5MTM0NyA1OC45NTgyNzU4Yy42NDQxNjA4LjYyODQ4IDEuNjg4NTQ5NS42Mjg0OCAyLjMzMjcxMDMgMCAuMDAwMDA5NS0uMDAwMDA5LjAwMDAxODItLjAwMDAxOC4wMDAwMjc3LS4wMDAwMjVsNjAuNDI2MTA2NS01OC45NTgyNTA4YzIuNTc2NTgxLTIuNTEzOTggNi43NTQxNDItMi41MTQwNzQzIDkuMzMwODQtLjAwMDIxMDMuMDAwMDM3LjAwMDAzNTQuMDAwMDcyLjAwMDA3MDkuMDAwMTA3LjAwMDEwNjNsNjAuNDI5MDU2IDU4Ljk1ODM1NDhjLjY0NDE1OS42Mjg0NzkgMS42ODg1NDkuNjI4NDc5IDIuMzMyNzA5IDBsNjAuNDI4MDc5LTU4Ljk1NzE5MjVjMi41NzY2NC0yLjUxMzkyMzEgNi43NTQxOTktMi41MTM5MjMxIDkuMzMwODM5IDB6IiBmaWxsPSIjZmZmIiBmaWxsLXJ1bGU9Im5vbnplcm8iIHRyYW5zZm9ybT0idHJhbnNsYXRlKDk4IDE2MCkiLz48L2c+PC9zdmc+"
                  />
                </div>
                <div
                  class="sc-gsnTZi kmGPKa web3modal-provider-name"
                >
                  WalletConnect
                </div>
                <div
                  class="sc-dkzDqf flNBYJ web3modal-provider-description"
                >
                  Scan with WalletConnect to connect
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="MuiBox-root css-tkevr6"
      >
        <span
          class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
          role="progressbar"
          style="width: 120px; height: 120px;"
        >
          <svg
            class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </span>
      </div>
      ,
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-tkevr6"
    >
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
        role="progressbar"
        style="width: 120px; height: 120px;"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </span>
    </div>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`BondDialog renders 1`] = `
<div
  aria-hidden="true"
>
  ,
</div>
`;
